src/main/java/com/kodilla/ecommercee/dto/ProductDto.java

```diff
@@ -15,5 +16,4 @@
     private String description;
     private BigDecimal price;
     private boolean availability;
-    private long groupId;
```

Comment:
próbowałeś złapać ten null przez zwykłego if'a?

Reply:
Próbowałem. Tzn. trzeba by było w metodzie mappera mapToProductDto(), stworzyć nowy obiekt ProductDto, a potem setterami dodawać mu wartości product'u. I wtedy można by zastosować warunek if - i w przypadku nulla np. wyświetlać "0" jako id grupy produktów. Tylko nie wiem, czy to jest poprawne rozwiązanie. Postaram się dzisiaj wymyślić może coś lepszego. Bo jednak dobrze by było, aby zwracana była informacja o grupie produktów, do której ten produkt należy. 

Ten Null Pointer Exception jest chyba związany z tym, że aplikacja nie może wyświetlić nulla jeśli pobiera taką wartość z bazy danych dla typu obiektowego - coś w tym stylu.

Reply:
Taki mam pomysł, żeby może ujednolicić pole groupId z typem, co jest w encji, czyli zamiast long niech to będzie ProductGroupDto i może wstrzyknąć zależność przez Autowire do klasy ProductService? Wtedy będzie można mapować te same typy ProductGroup i ProductGroupDto, co sądzisz?

Reply:
Pokombinuję jak wrócę do domu :)